lib/scheduler.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('fs');
+const path = require('path');
+const tempDir = require('temp-dir');
+
+const buildPath = projectDir => path.join(tempDir, 'ava', path.basename(projectDir));
+const FILE_NAME_FAILING_TEST = 'failing-test.json';
+
+module.exports.prepareTestResultsFile = projectDir => {
+	const temporaryResultsDir = buildPath(projectDir);
+	try {
+		fs.statSync(temporaryResultsDir);
+	} catch {
+		fs.mkdirSync(temporaryResultsDir, {recursive: true});
```

Comment:
IIRC this doesn't fail if the directory exists, so the `statSync` shouldn't be necessary.